File: 1. Learn the Basics/1. Pattern/C++/19.cpp
/*

**********
****  ****
***    ***
**      **
*        *
*        *
**      **
***    ***
****  ****
**********
*/

#include <bits/stdc++.h>
using namespace std;

void upperSymmetry(int n){
    int inital_spaces = 0;
    for(int i=0;i<n;i++){
        for(int j=1;j<=n-i;j++){
            cout << "*";
        }
        for(int j=0;j<inital_spaces;j++){
            cout << " ";
        }
        for(int j=1;j<=n-i;j++){
            cout << "*";
        }
        inital_spaces += 2;
        cout <<endl;
    }

}
void lowerSymmerty(int n){
    for(int i=1;i<=n;i++){
        for(int j=1;j<=i;j++){
            cout << "*";
        }
        for(int j=1;j<=2*(n-i);j++){
            cout << " ";
        }
        for(int j=i;j>0;j--){
            cout << "*";
        }
        cout << endl;
    }

}


int main() {
	// your code goes here
	int n;
	cin >> n;
	upperSymmetry(n);
	lowerSymmerty(n);
	return 0;
}

File: 1. Learn the Basics/1. Pattern/Python/19.py
# **********
# ****  ****
# ***    ***
# **      **
# *        *
# *        *
# **      **
# ***    ***
# ****  ****
# **********

def upperSymmetry(n):
    inital_spaces = 0
    for i in range(0,n):
        for j in range(1,n-i+1):
            print("*",end="")
        for j in range(0,inital_spaces):
            print(" ",end="")
        for j in range(1,n-i+1):
            print("*",end="")
        inital_spaces = inital_spaces+2
        print("\r")

def lowerSymmetry(n):
    for i in range(1,n+1):
        for j in range(1,i+1):
            print("*",end="")
        for j in range(1,2*(n-i)+1):
            print(" ",end="")
        for j in range(i,0,-1):
            print("*",end="")
        print("\r")
        

n = int(input())
upperSymmetry(n)
lowerSymmetry(n)